m3u8/reader.go

```diff
@@ -163,6 +163,52 @@ func (p *MediaPlaylist) SetVersion(ver uint8) {
 	p.ver = ver
 }
 
+func (p *MediaPlaylist) LastSegIndex() uint64 {
```

Comment:
I think this method deserves a doc string given that it is not trivial to understand what happens in cases like skippedSegs.